tests/baselines/reference/variadicTuples1.errors.txt

```diff
@@ -496,4 +499,20 @@ tests/cases/conformance/types/tuple/variadicTuples1.ts(354,26): error TS2322: Ty
     
     callApi(getUser);
     callApi(getOrgUser);
+    
+    // Repro from #40235
+    
+    type Numbers = number[];
+    type Unbounded = [...Numbers, boolean];
+                      ~~~~~~~~~~
+!!! error TS1256: A rest element must be last in a tuple type.
+    const data: Unbounded = [false, false];
+    
+    type U1 = [string, ...Numbers, boolean];
+                       ~~~~~~~~~~
+!!! error TS1256: A rest element must be last in a tuple type.
+    type U2 = [...[string, ...Numbers], boolean];
+               ~~~~~~~~~~~~~~~~~~~~~~~
+!!! error TS1256: A rest element must be last in a tuple type.
+    type U3 = [...[string, number], boolean];

```

Comment:
Not sure if I get this right, does this mean we still have (or bring back) the constraint that a rest element must be at last in a tuple?
In [4.0 doc](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html), I can see the below update:
> The second change is that rest elements can occur anywhere in a tuple - not just at the end!
> Previously, TypeScript would issue an error like the following:
> But with TypeScript 4.0, this restriction is relaxed.

Reply:
@ahejlsberg